lib/src/lexer.c

```diff
@@ -233,8 +238,25 @@ static void ts_lexer__mark_end(TSLexer *_self) {
 
 static uint32_t ts_lexer__get_column(TSLexer *_self) {
   Lexer *self = (Lexer *)_self;
+  
+  uint32_t goal_byte = self->current_position.bytes;
+  
   self->did_get_column = true;
-  return self->current_position.extent.column;
+  self->current_position.bytes -= self->current_position.extent.column;
+  self->current_position.extent.column = 0;
```

Comment:
One thing I worry about: at the very start of the file, could this possibly be moved prior to the BOM and count the BOM as a codepoint?

Reply:
I think that it could do that, but that would be ok. Currently, we lexer does count the BOM as a code point; it just automatically skips it as if it were leading whitespace.